angularjs-portal-home/src/main/webapp/my-app/layout/controllers.js

```diff
@@ -171,6 +141,41 @@ define(['angular', 'jquery'], function(angular, $) {
       vm.init();
     }])
 
+  .controller('RemoveWidgetController', ['$scope', 'layoutService',
+    '$sessionStorage', function($scope, layoutService,
+      $sessionStorage) {
+        var vm = this;
+        /**
+         * Remove widget from home layout
+         * @param fname
+         */
+        vm.removePortlet = function removePortletFunction(fname) {
+          layoutService.removeFromHome(fname).success(function() {
+            $scope.$apply(function(request, text) {
```

Comment:
I'm not sure `$scope.$apply` is needed here.
AFAICT All the operations all the operations are synchronous, and exception handling could be managed by the wrapping promise.
it appears be used to indirectly trigger `$scope.$digest`, which can be called directly.